### PR TITLE
Not create segment for empty table.

### PR DESCRIPTION
**Pull request by JinheLin**
### What problem does this PR solve?

Issue Number: ref #7595

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp [hunk 1]
```diff
@@ -1849,5 +1785,78 @@ String DeltaMergeStore::getLogTracingId(const DMContext & dm_ctx)
         return fmt::format("table_id={}", physical_table_id);
     }
 }
+
+std::pair<SegmentPtr, bool> DeltaMergeStore::getSegmentByStartKeyInner(const RowKeyValueRef & start_key)
+{
+    std::shared_lock lock(read_write_mutex);
+    auto seg_it = segments.upper_bound(start_key);
+    return seg_it != segments.end() ? std::pair{seg_it->second, segments.empty()} : std::pair{nullptr, segments.empty()};
+}
```
Comment (Lloyd-Pottiger) [inline dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp:1794]: seems only called by `DeltaMergeStore::getSegmentByStartKey`, maybe no need to abstract to a new function?

**Review by Lloyd-Pottiger — APPROVED** [review]

**Review by lidezhu — APPROVED** [review]
LGTM

**Pull request merged**